buildifier/utils/utils.go

```diff
@@ -115,13 +116,39 @@ func SplitFilePath(filename string) (workspaceRoot, pkg string) {
 	return workspaceRoot, pkg
 }
 
+// getFileReader returns a *FileReader object that reads files from the local
+// filesystem if the workspace root is known.
+func getFileReader(workspaceRoot string) *warn.FileReader {
+	if workspaceRoot == "" {
+		return nil
+	}
+
+	readFile := func(filename string) ([]byte, error) {
+	    // Use OS-specific path separators
+        filename = strings.ReplaceAll(filename, "/", string(os.PathSeparator))
+		path := filepath.Join(workspaceRoot, filename)
+
+		data, err := ioutil.ReadFile(path)
+		if err != nil {
+			return nil, err
+		}
+		return data, nil
+	}
+
+	fileReader := &warn.FileReader{}
+	fileReader.Init(readFile)
```

Comment:
If Init is always needed, just name a method

warn.NewFileReader

that does what Init does

Reply:
Done.